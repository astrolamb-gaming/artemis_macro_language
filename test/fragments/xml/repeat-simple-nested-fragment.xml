<mission_data version="1.0">
  <ranges>
    <range name="AllShips">
      <item ship="Artemis" sideValue="10" />
      <item ship="Intrepid" sideValue="11" />
      <item ship="Aegis" sideValue="4" />
      <item ship="Horatio" sideValue="5" />
      <item ship="Excalibur" sideValue="6" />
      <item ship="Hera" sideValue="7" />
      <item ship="Ceres" sideValue="8" />
      <item ship="Diana" sideValue="9" />
    </range>

    <range name="Eggs">
      <item egg="egg1" seconds="25" playerCount="0" x="50000" y="10" z="5000" />
      <item egg="egg2" seconds="30" playerCount="3.0" x="49000" y="3" z="49000" />
      <item egg="egg3" seconds="35" playerCount="4.0" x="51000" y="3" z="51000" />
      <item egg="egg4" seconds="35" playerCount="6.0" x="49000" y="3" z="5100" />
      <item egg="egg5" seconds="35" playerCount="7.0" x="51000" y="3" z="49000" />
    </range>
  </ranges>

  <start>
    <repeat range="Eggs">
      <set_variable name="${egg}" value="1.0"/>
      <set_timer name="${egg}_Timer" seconds="${seconds}"/>
    </repeat>
    <repeat range="AllShips">
      <set_variable name="${ship}_Cargo" value="0.0"/>
    </repeat>
  </start>

  <repeat range="AllShips" as="Ship">
    <repeat range="Eggs" as="Egg" angle="180" >
      <event name="${Ship.ship} to ${Egg.egg}">
        <set_relative_position name2="${Ship.ship}" distance="39000" angle="${angle}" name1="${Egg.egg}"/>
      </event>
    </repeat>
  </repeat>
</mission_data>