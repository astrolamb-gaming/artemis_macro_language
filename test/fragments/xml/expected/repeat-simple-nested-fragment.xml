<mission_data version="1.0">
    <start>
        <set_variable name="egg1" value="1.0"/>
        <set_timer name="egg1_Timer" seconds="25"/>
        <set_variable name="egg2" value="1.0"/>
        <set_timer name="egg2_Timer" seconds="30"/>
        <set_variable name="egg3" value="1.0"/>
        <set_timer name="egg3_Timer" seconds="35"/>
        <set_variable name="egg4" value="1.0"/>
        <set_timer name="egg4_Timer" seconds="35"/>
        <set_variable name="egg5" value="1.0"/>
        <set_timer name="egg5_Timer" seconds="35"/>
        <set_variable name="Artemis_Cargo" value="0.0"/>
        <set_variable name="Intrepid_Cargo" value="0.0"/>
        <set_variable name="Aegis_Cargo" value="0.0"/>
        <set_variable name="Horatio_Cargo" value="0.0"/>
        <set_variable name="Excalibur_Cargo" value="0.0"/>
        <set_variable name="Hera_Cargo" value="0.0"/>
        <set_variable name="Ceres_Cargo" value="0.0"/>
        <set_variable name="Diana_Cargo" value="0.0"/>
    </start>
    <event name="Artemis to egg1">
        <set_relative_position name2="Artemis" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Artemis to egg2">
        <set_relative_position name2="Artemis" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Artemis to egg3">
        <set_relative_position name2="Artemis" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Artemis to egg4">
        <set_relative_position name2="Artemis" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Artemis to egg5">
        <set_relative_position name2="Artemis" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Intrepid to egg1">
        <set_relative_position name2="Intrepid" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Intrepid to egg2">
        <set_relative_position name2="Intrepid" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Intrepid to egg3">
        <set_relative_position name2="Intrepid" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Intrepid to egg4">
        <set_relative_position name2="Intrepid" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Intrepid to egg5">
        <set_relative_position name2="Intrepid" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Aegis to egg1">
        <set_relative_position name2="Aegis" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Aegis to egg2">
        <set_relative_position name2="Aegis" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Aegis to egg3">
        <set_relative_position name2="Aegis" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Aegis to egg4">
        <set_relative_position name2="Aegis" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Aegis to egg5">
        <set_relative_position name2="Aegis" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Horatio to egg1">
        <set_relative_position name2="Horatio" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Horatio to egg2">
        <set_relative_position name2="Horatio" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Horatio to egg3">
        <set_relative_position name2="Horatio" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Horatio to egg4">
        <set_relative_position name2="Horatio" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Horatio to egg5">
        <set_relative_position name2="Horatio" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Excalibur to egg1">
        <set_relative_position name2="Excalibur" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Excalibur to egg2">
        <set_relative_position name2="Excalibur" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Excalibur to egg3">
        <set_relative_position name2="Excalibur" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Excalibur to egg4">
        <set_relative_position name2="Excalibur" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Excalibur to egg5">
        <set_relative_position name2="Excalibur" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Hera to egg1">
        <set_relative_position name2="Hera" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Hera to egg2">
        <set_relative_position name2="Hera" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Hera to egg3">
        <set_relative_position name2="Hera" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Hera to egg4">
        <set_relative_position name2="Hera" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Hera to egg5">
        <set_relative_position name2="Hera" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Ceres to egg1">
        <set_relative_position name2="Ceres" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Ceres to egg2">
        <set_relative_position name2="Ceres" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Ceres to egg3">
        <set_relative_position name2="Ceres" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Ceres to egg4">
        <set_relative_position name2="Ceres" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Ceres to egg5">
        <set_relative_position name2="Ceres" distance="39000" angle="180" name1="egg5"/>
    </event>
    <event name="Diana to egg1">
        <set_relative_position name2="Diana" distance="39000" angle="180" name1="egg1"/>
    </event>
    <event name="Diana to egg2">
        <set_relative_position name2="Diana" distance="39000" angle="180" name1="egg2"/>
    </event>
    <event name="Diana to egg3">
        <set_relative_position name2="Diana" distance="39000" angle="180" name1="egg3"/>
    </event>
    <event name="Diana to egg4">
        <set_relative_position name2="Diana" distance="39000" angle="180" name1="egg4"/>
    </event>
    <event name="Diana to egg5">
        <set_relative_position name2="Diana" distance="39000" angle="180" name1="egg5"/>
    </event>
</mission_data>